book/src/DEVELOPER_README.md

```diff
@@ -1,5 +1,90 @@
+things have always been done". But we do this to ensure that all people can have their identities
+stored how they choose.
+
+For example, any user may change their name, display name and legal name at any time. Many
+applications will break as they primary key from name when this occurs. But this is the fault of the
+application. Name changes must be allowed. Our job as technical experts is to allow that to happen.
+
+We will never put a burden on the user to correct for poor designs on our part. For example, locking
+an account if it logs in from a different country unless the user logs in before hand to indicate
+where they are going. This makes the user responsible for a burden (changing the allowed login
+country) when the real problem is preventing bruteforce attacks - which can be technically solved in
+better ways that doesn't put administrative load to humans.
+
+### Correct and Simple
+
+As a piece of security sensitive software we must always put correctness first. All code must have
+tests. All developers must be able to run all tests on their machine and environment of choice.
+
+This means that the following must always work:
+
+```bash
+git clone ...
+cargo test
+```
+
+If a test or change would require extra requirements, dependencies, or preconfiguration, then we can
+no longer provide the above. Testing must be easy and accessible, else we won't do it, and that
+leads to poor software quality.
+
+The project must be simple. Any one should be able to understand how it works and why those
+decisions were made.
+
+### Hierachy of Controls
+
+When a possible risk arises we should always consider the [hierachy of controls]. In descedending
+order of priority
+
+- Elimination - eliminate the risk from existing
+- Substitution - replace the risk with something less dangerous
+- Engineering Controls - isolate the risk from causing harm
+- Administrative Controls - educate about the risk, add warnings
+- Personal Protection - document the risk
+
+[hierachy of controls]: https://en.wikipedia.org/wiki/Hierarchy_of_hazard_controls
+
+### Languages
+
+The core server will (for now) always be written in Rust. This is due to the strong type guarantees
+it gives, and how that can help raise the quality of our project.
+
+### Over-Configuration
+
+Configuration will be allowed, but only if it does not impact the statements above. Having
+configuration is good, but allowing too much (i.e. a scripting engine for security rules) can give
+deployments the ability to violate human first principles, which reflects badly on us.
+
+In addition every extra configuration item expands our testing matrix exponentially. We should
+optimally only offer one path that is correct for all users unless no other options are possible.
+
+All configuration items, must be constrained to fit within our principles so that every kanidm
+deployment, will always provide a positive experience to all people.
```

Comment:
```suggestion
deployment will aim to provide a positive experience to all people.
```

it's *impossible* to give everyone a positive experience, but we can balance the outcomes.